internal/operator-controller/applier/boxcutter.go

```diff
@@ -0,0 +1,225 @@
+	var (
+		currentRevision *ocv1.ClusterExtensionRevision
+		prevRevisions   []ocv1.ClusterExtensionRevision
+	)
+	if len(existingRevisions) > 0 {
+		maybeCurrentRevision := existingRevisions[len(existingRevisions)-1]
+
+		annotations := maybeCurrentRevision.GetAnnotations()
+		if annotations != nil {
+			if hash, ok := annotations[revisionHashAnnotation]; ok &&
+				hash == desiredHash {
+				currentRevision = &maybeCurrentRevision
+				prevRevisions = existingRevisions[0 : len(existingRevisions)-1] // previous is everything excluding current
+			}
+		}
+	}
+
+	if currentRevision == nil {
+		// all Revisions are outdated => create a new one.
+		prevRevisions = existingRevisions
+		revisionNumber := latestRevisionNumber(prevRevisions)
+		revisionNumber++
+
+		newRevision := desiredRevision
+		newRevision.Spec.Revision = revisionNumber
+		// newRevision.Spec.Previous
+		for _, prevRevision := range prevRevisions {
+			newRevision.Spec.Previous = append(newRevision.Spec.Previous, ocv1.ClusterExtensionRevisionPrevious{
+				Name: prevRevision.Name,
+				UID:  prevRevision.UID,
+			})
+		}
+
+		if err := controllerutil.SetControllerReference(ext, newRevision, bc.Scheme); err != nil {
+			return nil, fmt.Errorf("set ownerref: %w", err)
+		}
+		if err := bc.Client.Create(ctx, newRevision); err != nil {
+			return nil, fmt.Errorf("creating new Revision: %w", err)
+		}
+	}
+
+	// Delete archived previous revisions over revisionHistory limit
+	numToDelete := len(prevRevisions) - revisionHistoryLimit
+	slices.Reverse(prevRevisions)
+
+	for _, prevRev := range prevRevisions {
+		if numToDelete <= 0 {
+			break
+		}
+
+		if err := client.IgnoreNotFound(bc.Client.Delete(ctx, &prevRev)); err != nil {
+			return nil, fmt.Errorf("failed to delete revision (history limit): %w", err)
+		}
+		numToDelete--
+	}
+
+	// TODO: Read status from revision.
+
+	return plain.Objects, nil
+}
+
+// computeSHA256Hash returns a sha236 hash value calculated from object.
+func computeSHA256Hash(obj any) string {
+	hasher := sha256.New()
+	deepHashObject(hasher, obj)
+	return hex.EncodeToString(hasher.Sum(nil))
+}
+
+// deepHashObject writes specified object to hash using the spew library
+// which follows pointers and prints actual values of the nested objects
+// ensuring the hash does not change when a pointer changes.
+func deepHashObject(hasher hash.Hash, objectToWrite any) {
+	hasher.Reset()
+
+	printer := spew.ConfigState{
+		Indent:         " ",
+		SortKeys:       true,
+		DisableMethods: true,
+		SpewKeys:       true,
+	}
+	if _, err := printer.Fprintf(hasher, "%#v", objectToWrite); err != nil {
+		panic(err)
+	}
+}
+
+type revisionAscending []ocv1.ClusterExtensionRevision
+
+func (a revisionAscending) Len() int      { return len(a) }
+func (a revisionAscending) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a revisionAscending) Less(i, j int) bool {
+	iObj := a[i]
+	jObj := a[j]
+
+	return iObj.Spec.Revision < jObj.Spec.Revision
+}
```

Comment:
Nit: might be able to avoid the type definition and associated boilerplate with `slices.SortFunc`?